src/mips64/cpu.rs

```diff
@@ -567,7 +862,10 @@ impl Cpu {
             self.ctx.tight_exit = false;
             while let Some(op) = iter.next() {
                 self.ctx.pc += 4;
-                self.op(op);
+                match self.op(op) {
```

Comment:
quick comment, this adds a cmp+branch on any opcode. This loop here is hot as fire, I think it's better to let the opcode implementation call `exception()` whenever an exception occurs, rather than propagate a Result and then having to check it for every opcode.